beluga_tutorial/src/main.cpp

```diff
@@ -0,0 +1,335 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cmath>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+
+#include <range/v3/action.hpp>
+#include <range/v3/algorithm/any_of.hpp>
+#include <range/v3/algorithm/min_element.hpp>
+#include <range/v3/numeric/accumulate.hpp>
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/generate.hpp>
+#include <range/v3/view/take_exactly.hpp>
+
+#include <yaml-cpp/yaml.h>
+#include <beluga/beluga.hpp>
+
+struct TutorialParams {
+  /// Load the landmark map from a file.
+  /**
+   * If true, the landmark map is loaded from a file;
+   * if false, a random map is generated.
+   */
+  bool load_landmark_map{false};
+
+  /// Save the landmark map to a file.
+  bool save_landmark_map{false};
+
+  /// Size of the 1D map in (m)
+  /**
+   * In the simulation, the 1D map is continuous,
+   * but the limit is defined as an integer to simplify data visualization.
+   */
+  int map_size{100};
+
+  /// Maximum number of doors (or landmarks) in the map.
+  int number_of_doors{33};
+
+  /// Fixed number of particles used by the algorithm.
+  int number_of_particles{200};
+
+  /// Number of simulation cycles.
+  int number_of_cycles{100};
+
+  /// Robot's initial pose in meters (m).
+  double initial_pose{0.0};
+
+  /// Standard deviation of the robot's initial pose.
+  /**
+   * Represents the uncertainty of the robot's initial pose.
+   */
+  double initial_pose_sd{1.0};
+
+  /// Delta time in seconds (s).
+  double dt{1.0};
+
+  /// Translation velocity in meters per second (m/s) of the robot in 1D.
+  double velocity{1.0};
+
+  /// Translation standard deviation.
+  /**
+   * Represents the robot's translation noise due to drift and slippage.
+   */
+  double translation_sd{1.0};
+
+  /// Sensor range of view in meters (m)
+  double sensor_range{2.0};
+
+  /// Sensor model sigma
+  /**
+   * Represents the precision of the modeled sensor.
+   */
+  double sensor_model_sigam{1.0};
+
+  /// Minimum particle weight.
+  /**
+   * Used to keep all particles "alive" in the reweight step.
+   */
+  double min_particle_weight{0.08};
+
+  /// Name of the dataset file.
+  /**
+   * The dataset file is used to save the data produced by the simulation for posterior analisys.
+   */
+  std::string dataset_file_name{"dataset.yaml"};
+
+  // Name of the landmark_map to save or load from.
+  std::string landmark_map_file_name{"landmark_map.yaml"};
+
+  /// Bags folder path to save the simulation datasets.
+  std::string bags_folder_path{"./src/beluga/beluga_tutorial/bags/"};
```

Comment:
Should we keep a relative path here? :thinking:

Reply:
Actually I left it there to create that path in case it doesn't exist, but you're right, I'll do it another way.